overlay/benchwrapper.py

```diff
@@ -0,0 +1,92 @@
+import re
+import subprocess
+import sys
+
+test_cpus = int(sys.argv[1])
+commands_to_run = sys.argv[2:]
+
+
+def call_systemctl(args):
+    return subprocess.check_output(["systemctl"] + args).decode()
+
+
+def call_systemd_run(args):
+    p = subprocess.Popen(["systemd-run"] + args)
+    p.communicate()
+
+
+def get_allowed_cpus():
```

Comment:
Unless set explicitly, the system slice does not have a allowedcpus setting.

That would result in this function returning an empty set when the wrapper has not been run after a reboot

Reply:
Also, this script will allow to request half of the available cpus on each invocation and then leave e.g. 1/2 of the total cpus after the first invocation, then only 1/4 of the total cpus after the second invocation n the worst case. 
This could result in the system being left with only one cpu.

Maybe leaving a fixed amount of cpus for the system and splitting up the other half between the requested slices would be better (on a first come first serve basis to preserve consistency - dynamic rescheduling would only lead to unpredictable runtimes)? :thinking:

Reply:
> Unless set explicitly, the system slice does not have a allowedcpus setting.

Good catch, perhaps it should fallback to `lscpu` in such case

Reply:
> Maybe leaving a fixed amount of cpus for the system and splitting up the other half between the requested slices would be better (on a first come first serve basis to preserve consistency - dynamic rescheduling would only lead to unpredictable runtimes)?

Sounds good to me